src/libmongoc/doc/index.rst

```diff
@@ -26,3 +26,5 @@ It depends on :doc:`libbson <bson:index>` to generate and parse BSON documents,
    guides
    api
    application-performance-monitoring
+
+   genindex
```

Comment:
Nice! But also exposes an out-of-place inclusion of the `mongoc_kms_credentials_provider_callback_fn` code block from the `mongoc_client_encryption_opts_set_kms_credential_provider_callback` page in the index. Recommend removing this entry from the index if able.

Reply:
Ideally, everything would be documented with directives that generate appropriate index entries. Since that lone `typedef` is the only one (yet) using a directive, it generates an index entry. Maybe in the future that can be changed. Until then, the `:noindexentry:` option will suffice.